internal/service/handlers.go

```diff
@@ -155,11 +155,52 @@ type DeleteHandler interface {
 	Delete(ctx context.Context, request *DeleteRequest) (response *DeleteResponse, err error)
 }
 
+// Handlers for configmap process.
+type ConfigMapChangeNotifyHandler interface {
+	//notification from db to application about db entry changes
+	//currently assume the notification is granular to indivial entry
+	ConfigMapChangeNotify(ctx context.Context, entry_key string, add bool)
+}
+
+type ConfigMapReadEntryHandler interface {
+	//notification from db to application about db entry changes
+	//currently assume the notification is granular to indivial entry
+	ConfigMapReadEntry(ctx context.Context, entry_key string) (value string, err error)
+}
+
+type ConfigMapAddEntryHandler interface {
+	//add db entry
+	//Currently assume as key value pair
+	ConfigMapAddEntry(ctx context.Context, entry_key string, value string) (err error)
+}
+
+type ConfigMapDeleteEntryHandler interface {
+	//delete db entry
+	ConfigMapDeleteEntry(ctx context.Context, entry_key string) (err error)
+}
+
+type RecoveryFromConfigMapHandler interface {
+	//The handler read to DB and build/recovery datastructures in memory
+	RecoveryFromConfigMap(ctx context.Context) (err error)
+}
+
+type WatchConfigMapHandler interface {
+	//The handler read to DB and build/recovery datastructures in memory
+	WatchConfigMap(ctx context.Context) (err error)
+}
```

Comment:
In general it is good practice in Go to make interfaces small, but I'd say that this is taking that too far. These methods are closely related so I think they belong in a single interface. Also the names of the methods are two configmap/kubernetes specific. What if we want to make an implementation that uses a relational database? Can you try to come up with something more generic?

Reply:
Addressed

Reply:
Made the changes.